crates/solvers/src/boundary/naive.rs

```diff
@@ -0,0 +1,239 @@
+            AmmOrderExecution,
+            ConstantProductOrder,
+            Exchange,
+            LimitOrder,
+            LimitOrderId,
+            LiquidityOrderId,
+            SettlementHandling,
+        },
+        settlement::{external_prices::ExternalPrices, SettlementEncoder},
+        solver::naive_solver::multi_order_solver,
+    },
+    std::sync::{Arc, Mutex},
+};
+
+pub fn solve(
+    orders: &[&order::Order],
+    liquidity: &liquidity::Liquidity,
+) -> Option<solution::Solution> {
+    let pool = match &liquidity.state {
+        liquidity::State::ConstantProduct(pool) => pool,
+        _ => return None,
+    };
+
+    let boundary_orders = orders
+        .iter()
+        .map(|order| LimitOrder {
+            id: match order.class {
+                order::Class::Market => LimitOrderId::Market(OrderUid(order.uid.0)),
+                order::Class::Limit => LimitOrderId::Limit(OrderUid(order.uid.0)),
+                order::Class::Liquidity => {
+                    LimitOrderId::Liquidity(LiquidityOrderId::Protocol(OrderUid(order.uid.0)))
+                }
+            },
+            sell_token: order.sell.token.0,
+            buy_token: order.buy.token.0,
+            sell_amount: order.sell.amount,
+            buy_amount: order.buy.amount,
+            kind: match order.side {
+                order::Side::Buy => OrderKind::Buy,
+                order::Side::Sell => OrderKind::Sell,
+            },
+            partially_fillable: order.partially_fillable,
+            solver_fee: order.fee().amount,
+            settlement_handling: Arc::new(OrderHandler {
+                order: Order {
+                    metadata: OrderMetadata {
+                        uid: OrderUid(order.uid.0),
+                        class: match order.class {
+                            order::Class::Market => OrderClass::Market,
+                            order::Class::Limit => OrderClass::Limit(LimitOrderClass {
+                                surplus_fee: Some(order.fee().amount),
+                                ..Default::default()
+                            }),
+                            order::Class::Liquidity => OrderClass::Liquidity,
+                        },
+                        solver_fee: order.fee().amount,
+                        ..Default::default()
+                    },
+                    data: OrderData {
+                        sell_token: order.sell.token.0,
+                        buy_token: order.buy.token.0,
+                        sell_amount: order.sell.amount,
+                        buy_amount: order.buy.amount,
+                        fee_amount: order.fee().amount,
+                        kind: match order.side {
+                            order::Side::Buy => OrderKind::Buy,
+                            order::Side::Sell => OrderKind::Sell,
+                        },
+                        partially_fillable: order.partially_fillable,
+                        ..Default::default()
```

Comment:
Don't we need to initialize all fields for the order to be correctly encoded?

Reply:
Maybe I misunderstood how this works. Since the actual encoding will be done later by the driver we probably only have to initialize the fields that matter for the naive algorithm. And I guess that is probably also why we don't care about the difference between fee_amount and solver_fee?

Reply:
> Since the actual encoding will be done later by the driver we probably only have to initialize the fields that matter for the naive algorithm. And I guess that is probably also why we don't care about the difference between fee_amount and solver_fee?

Exactly. I will add a comment blob on top of the type mapping.

Reply:
6700e0fb429776121c5399e4388c24e763c29367